From aa88bb63819a70ed5fb70fe9690fb5f58497e42c Mon Sep 17 00:00:00 2001
From: Sumit Singh <sumsingh@nvidia.com>
Date: Fri, 18 Apr 2014 20:00:36 +0530
Subject: [PATCH 17/27] asm-generic: relaxed.h: defined relaxed.h

Defined a new header file relaxed.h, which contains basic
macros which will be used for improving power efficiency for
arm64.

bug 1440421

Change-Id: I5ae7503afdfbaa951827bbf466d8ddccf444f558
Signed-off-by: Sumit Singh <sumsingh@nvidia.com>
Reviewed-on: http://git-master/r/398315
(cherry picked from commit dd434aeb1afea1d9ebce1099fb6ecfa7c6c762c6)
Reviewed-on: http://git-master/r/422203
Reviewed-by: Automatic_Commit_Validation_User
GVS: Gerrit_Virtual_Submit
Reviewed-by: Bharat Nihalani <bnihalani@nvidia.com>
Signed-off-by: Jesse Chan <jc@linux.com>
Signed-off-by: Michele Beccalossi <beccalossi.michele@gmail.com>
---
 include/asm-generic/relaxed.h | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)
 create mode 100644 include/asm-generic/relaxed.h

diff --git a/include/asm-generic/relaxed.h b/include/asm-generic/relaxed.h
new file mode 100644
index 00000000000..37dc310a4b4
--- /dev/null
+++ b/include/asm-generic/relaxed.h
@@ -0,0 +1,30 @@
+/*
+ * include/asm-generic/relaxed.h
+ *
+ * Copyright (c) 2014 NVIDIA Corporation. All rights reserved.
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _ASM_GENERIC_RELAXED_H_
+#define _ASM_GENERIC_RELAXED_H_
+
+#ifndef cpu_relaxed_read_short
+#define cpu_relaxed_read_short(p)	(*(p))
+#endif
+
+#ifndef cpu_relaxed_read
+#define cpu_relaxed_read(p)		(*(p))
+#endif
+
+#ifndef cpu_relaxed_read_long
+#define cpu_relaxed_read_long(p)	(*(p))
+#endif
+
+#endif /*_ASM_GENERIC_RELAXED_H_*/
-- 
2.11.0

